lib/OperatingSystems/DefaultOs.php

```diff
@@ -143,19 +143,19 @@ public function getNetworkInfo() {
 	 * @return string
 	 */
 	public function getNetworkInterfaces() {
-		$interfaces = glob('/sys/class/net/*');
+		$interfaces = array_diff(scandir("/sys/class/net"), ['.', '..']);  // remove dot directories
 		$result = [];
 
 		foreach ($interfaces as $interface) {
 			$iface              = [];
-			$iface['interface'] = basename($interface);
+			$iface['interface'] = $interface;
 			$iface['mac']       = shell_exec('ip addr show dev ' . $iface['interface'] . ' | grep "link/ether " | cut -d \' \' -f 6  | cut -f 1 -d \'/\'');
 			$iface['ipv4']      = shell_exec('ip addr show dev ' . $iface['interface'] . ' | grep "inet " | cut -d \' \' -f 6  | cut -f 1 -d \'/\'');
 			$iface['ipv6']      = shell_exec('ip -o -6 addr show ' . $iface['interface'] . ' | sed -e \'s/^.*inet6 \([^ ]\+\).*/\1/\'');
 			if ($iface['interface'] !== 'lo') {
 				$iface['status'] = shell_exec('cat /sys/class/net/' . $iface['interface'] . '/operstate');
 				$iface['speed']  = shell_exec('cat /sys/class/net/' . $iface['interface'] . '/speed');
-				if ($iface['speed'] !== '') {
+				if ($iface['speed'] !== '' and substr($iface['speed'], 0, 1) != "-") {
```

Comment:
```suggestion
				if ($iface['speed'] !== '' && $iface['speed'] !== '-1') {
```

Reply:
Your suggestion did not work. Have tested it with ` !== '-1'`.
Content of /sys/class/net/*/speed results in "`-1\x0a`".
```
root@nextcloud2:/sys/class/net$ hexdump -C ens3/speed
00000000  2d 31 0a                                          |-1.|
```
I hope an extra trim is ok as solution.

I've spent hours on git now. I'm not yet git aware. Sorry for the mess and the force push. Got DCO fixed at least.

Reply:
Thanks